### ISSUE TITLE
Don't ask user to run in background on restart after clicking upgrade

### ISSUE THREAD

**Issue by lrettig** [spacemeshos/smapp#1457]
### Environment

* Smapp version: 1.0.17

### Describe the bug

When Smapp autodetects a new release, it displays a message at the bottom left corner of the screen with a link that allows the user to download the new release. Once the download is complete, it triggers a restart, and the user gets the usual prompt on quitting, prompting them whether they want to run in the background or quit.

I don't think the user should receive this prompt when upgrading. In particular, if go-spacemesh itself is being updated, then there's no choice and the node will have to be shut down. I get confused every time I see this message.

Could we replace it with a message that says something like, "Restart now? Note that if your node is about to publish a proposal, you might miss a small amount of rewards. You can complete the upgrade by restarting later if you prefer." If the user chooses to postpone the restart, there should be a persistent message (in place of the upgrade message) reminding them that they need to restart to complete the upgrade. If they try to close the app, they should be presented with the same message again (rather than the usual "run in background" message).

*Ideally* Smapp would also warn the user if the smesher is eligible to produce a proposal in, say, the next 5 layers, in which case it would make sense to postpone the restart.

### Steps to reproduce

Click Upgrade when an upgrade is available
